File: FrameworkExample/Objective-C/VponSdkSampleObjC/VponSdkSampleObjC/VponMenuTableViewController.m
//
//  VponMenuTableViewController.m
//  VponSdkSampleObjC
//
//  Created by EricChien on 2017/7/18.
//  Copyright © 2017年 Soul. All rights reserved.
//

#import "VponMenuTableViewController.h"

#import "VponSdkVastInTableRepeatViewController.h"
#import "VponSdkVastInTableViewController.h"

@interface VponMenuTableViewController ()

@property (strong, nonatomic) NSDictionary *data;

@property (strong, nonatomic) NSArray *sortKeys;

@end

@implementation VponMenuTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"Menu";
    
    _data = [[NSDictionary alloc] initWithObjectsAndKeys:
                 @"goVponSdkBannerViewController", @"SDK - Banner",
                 @"goVponSdkInterstitialViewController", @"SDK - Interstitial",
                 @"goVponSdkNativeViewController", @"SDK - Native",
                 @"goVponSdkVastCustomAdViewController", @"SDK - VastCustomAd",
                 @"goVponSdkVastInScrollViewController", @"SDK - VastInScroll",
                 @"goVponSdkVastInTableViewController", @"SDK - VastInTable",
                 @"goVponSdkVastInTableRepeatViewController", @"SDK - VastInTableRepeat",  nil];
    
    _sortKeys = [_data.allKeys sortedArrayUsingDescriptors:@[
        [NSSortDescriptor sortDescriptorWithKey:@"self" ascending:YES]
    ]];
}

- (void) viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [self.navigationController setNavigationBarHidden:NO animated:YES];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return _data.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"reuseIdentifier" forIndexPath:indexPath];
    cell.textLabel.text = [_sortKeys objectAtIndex:indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    NSString *key = [_sortKeys objectAtIndex:indexPath.row];
    [self performSegueWithIdentifier:[_data objectForKey:key] sender:nil];
}

@end

File: FrameworkExample/Swift/VponSdkSampleSwift/VponSdkSampleSwift/VponMenuTableViewController.swift
//
//  VponMenuTableViewController.swift
//  VponDFPSampleSwift
//
//  Created by EricChien on 2017/6/12.
//  Copyright © 2017年 Soul. All rights reserved.
//

import UIKit

class VponMenuTableViewController: UITableViewController {
    
    let data = ["SDK - Banner" : "goVponSdkBannerViewController",
                "SDK - Interstitial" : "goVponSdkInterstitialViewController",
                "SDK - Native" : "goVponSdkNativeViewController",
                "SDK - VastCustomAd": "goVponSdkVastCustomAdViewController",
                "SDK - VastInScroll": "goVponSdkVastInScrollViewController",
                "SDK - VastInTable": "goVponSdkVastInTableViewController",
                "SDK - VastInTableRepeat": "goVponSdkVastInTableRepeatViewController"]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Menu"
    }
    
    override func viewWillAppear(_ animated: Bool) {
        self.navigationController?.setNavigationBarHidden(false, animated: true)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    // MARK: - Table view data source

    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return data.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "reuseIdentifier", for: indexPath)

        let titles = data.keys.sorted()
        cell.textLabel?.text = titles[indexPath.row]

        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let identifiers = data.values.sorted()
        self.performSegue(withIdentifier: identifiers[indexPath.row], sender: nil)
    }
}
